src/DynamoCoreWpf/Views/Preview/WatchTree.xaml.cs

```diff
@@ -14,27 +14,71 @@ public partial class WatchTree : UserControl
     {
         private WatchViewModel _vm;
         private WatchViewModel prevWatchViewModel;
-        private readonly int defaultSize = 200;
-        private readonly int minSize = 100;
+        private readonly int defaultWidthSize = 200;
+        private readonly int defaultHeightSize = 200;
+        private readonly int minWidthSize = 100;
+        private readonly int minHeightSize = 38;
 
         public WatchTree()
         {
             InitializeComponent();
             this.Loaded += WatchTree_Loaded;
+            this.Unloaded += WatchTree_Unloaded;
+        }
+
+        private void WatchTree_Unloaded(object sender, RoutedEventArgs e)
+        {
+            _vm.PropertyChanged -= _vm_PropertyChanged;
         }
 
         void WatchTree_Loaded(object sender, RoutedEventArgs e)
         {
             _vm = this.DataContext as WatchViewModel;
+            _vm.PropertyChanged += _vm_PropertyChanged;            
+        }
+
+        private void _vm_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "IsCollection")
+            {
+                if (_vm.IsCollection)
+                {
+                    this.Height = defaultHeightSize;
+                }
+                else
+                {
+                    this.Height = minHeightSize;
+                }
+            }
+            else if (e.PropertyName == "Children")
+            {
+                if (_vm.Children != null)
+                {
+                    if (!_vm.Children[0].IsCollection)
+                    {
+                        double requiredWidth = (_vm.Children[0].NodeLabel.Length * 7.5);
```

Comment:
Why 7.5 ?
A comment at least

Reply:
this is the width factor for each character, it has been approved / and evidenced here https://github.com/DynamoDS/Dynamo/pull/12829, this current PR is focusing in the tests.